lib/content-services/content-node-selector/content-node-dialog.service.ts

```diff
@@ -45,6 +49,41 @@ export class ContentNodeDialogService {
         });
     }
 
+    /**
+     * Opens a lock node dialog
+     *
+     * @param contentEntry Node to lock
+     */
+    public openLockNodeDialog(contentEntry: MinimalNodeEntryEntity): Subject<string> {
+        const observable: Subject<string> = new Subject<string>();
+
+        if (this.contentService.hasPermission(contentEntry, PermissionsEnum.LOCK)) {
+            const dialogInstance = this.dialog.open(NodeLockDialogComponent, {
+                data: { node: contentEntry },
+                width: '400px'
+            });
+
+            dialogInstance.afterOpen().subscribe(() => {
+                observable.next('OPERATION.SUCCESS.NODE.LOCK_DIALOG_OPEN');
+            });
+
+            dialogInstance.componentInstance.error.subscribe((error) => {
```

Comment:
Instead of reaching the "component's guts", I think it would be better to pass down the error callback in the data object, just as we did in the "openCopyMoveDialog" method of the same file (select property of the data)